File: replace.py
import os

def replace_in_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # JS files .toFixed(2) -> .toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 })
    # but wait, some are like $ which is fine.
    
    # Let's just use string replacement carefully
    replacements = {
        '.toFixed(2)': \".toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 })\",
        'parseFloat(producto.precio).toLocaleString(\\'es-CO\\', { minimumFractionDigits: 0 })': 'parseFloat(producto.precio).toLocaleString(\\'es-CO\\', { minimumFractionDigits: 0, maximumFractionDigits: 0 })',
        '<%= prod.precio %>': \"<%= parseFloat(prod.precio).toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 }) %>\",
        '<%= producto.precio %>': \"<%= parseFloat(producto.precio).toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 }) %>\",
    }

    new_content = content
    for old, new in replacements.items():
        if 'data-precio=' not in old: # Don't replace data attributes! Wait, we need to be careful with <%= prod.precio %> inside data-precio=\"<%= prod.precio %>\".
            pass
            
    # Safer manual replacements:
    # 1. toFixed(2)
    new_content = new_content.replace('.toFixed(2)', \".toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 })\")
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(new_content)

for root, dirs, files in os.walk('.'):
    for file in files:
        if file.endswith('.js') or file.endswith('.ejs'):
            path = os.path.join(root, file)
            if 'node_modules' in path: continue
            replace_in_file(path)
print('Done!')

File: replace.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            if (!file.includes('node_modules') && !file.includes('.git')) {
                results = results.concat(walk(file));
            }
        } else {
            if (file.endsWith('.js') || file.endsWith('.ejs')) {
                results.push(file);
            }
        }
    });
    return results;
}

const files = walk(__dirname);

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let original = content;
    
    // Replace .toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 }) -> .toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 })
    content = content.replace(/\.toFixed\(2\)/g, ".toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 })");
    
    // Also format EJS templates
    // But be careful NOT to replace data-precio="<%= prod.precio %>" or <%= producto.precio %> where it's used as data!
    // We only replace >$<%= parseFloat(prod.precio).toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 }) %>< or >$<%= parseFloat(producto.precio).toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 }) %><
    content = content.replace(/>\$\<%= prod\.precio %\></g, ">$<%= parseFloat(prod.precio).toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 }) %><");
    content = content.replace(/>\$\<%= producto\.precio %\></g, ">$<%= parseFloat(producto.precio).toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 }) %><");
    content = content.replace(/\$\<%= prod\.precio %\>/g, "$<%= parseFloat(prod.precio).toLocaleString('es-CO', { minimumFractionDigits: 0, maximumFractionDigits: 0 }) %>");
    
    if (content !== original) {
        fs.writeFileSync(file, content, 'utf8');
        console.log('Updated', file);
    }
});
console.log('Done!');
